src/electron-main.ts

```diff
@@ -186,13 +182,42 @@ async function setupGlobals(): Promise<void> {
 
         // Could not load local config, this is expected in most cases.
     }
+}
+
+// Configure Electron Sentry and crashReporter using sentry.dsn in config.json if one is present.
+async function configureSentry(): Promise<void> {
+    await loadConfig();
+    const { dsn, environment } = global.vectorConfig.sentry || {};
+    if (dsn) {
+        console.log(`Enabling Sentry with dsn=${dsn} environment=${environment}`);
+        Sentry.init({
+            dsn,
+            environment,
+            // We don't actually use this IPC, but we do not want Sentry injecting preloads
+            ipcMode: Sentry.IPCMode.Classic,
+        });
+    }
+}
+configureSentry();
```

Comment:
this line feels a bit lost, here in the middle of all these function definitions. Maybe move it down to the other setup things around line 288?